File: ruby/bob/bob.rb
class Bob
  def hey(remark)
    remark.delete!("\n")
    case
    when ( remark == remark.upcase ) && ( remark.match(/[A-Z]/) )
      'Whoa, chill out!'
    when remark.nil?, remark.match(/^\s*$/)
      'Fine. Be that way!'
    when remark.match(/\?$/)
      'Sure.'
    else
      'Whatever.'
    end
  end
end

File: elixir/bob/bob.exs
defmodule Bob do
  def hey(input) do
    cond do
      Regex.match?(~r/\?$/, input) ->
        "Sure."
      String.strip(input) == "" ->
        "Fine. Be that way!"
      String.upcase(input) == input and Regex.match?(~r/[\p{L}]/, input) ->
        "Whoa, chill out!"
      true ->
        "Whatever."
    end
  end
end
